cogs/vxtwitconverter/eventsCore.py

```diff
@@ -1,49 +1,141 @@
+                "Here's a better alternative:\n",
+                "\n".join(vx_twit_links),
+            ]
+        )
+
+    @staticmethod
+    def _valid(message: Message):
+        """
+        Parameters
+        ----------
+        message: Discord input message object
+
+        Returns
+        -------
+            True if the message is from a human in a guild and contains video embeds
+            False otherwise
+        """
+
         # skips if the message is sent by any bot
         if message.author.bot:
-            return
+            return False
 
         # skips if message is in dm
         if isinstance(message.channel, channel.DMChannel):
+            return False
+
+        # skips if the message has no embeds
+        if not any(embed.video for embed in message.embeds):
+            return False
+
+        return True
+
+    async def _on_message_twit_replacer(self, message: Message):
+        self.logger.info("message")
+
+        if not self._valid(message):
             return
 
         if not await self.config.guild(message.guild).get_attr(KEY_ENABLED)():
-            self.logger.debug(
+            self.logger.info(
                 "VxTwit disabled for guild %s (%s), skipping", message.guild.name, message.guild.id
             )
             return
 
-        # skips if the message has no embeds
-        if not any(embed.video for embed in message.embeds):
+        # the actual code part
+        vx_twtter_urls = self._convert_to_vx_twitter_url(message.embeds)
+
+        # no changed urls detected
+        if not vx_twtter_urls:
             return
 
-        # the actual code part
-        vx_twit_links = [
-            result.replace("https://twitter.com", "https://vxtwitter.com")
-            for result in URLExtract().find_urls(message.content)
-            if "https://twitter.com" in result
+        # constructs the message and replies with a mention
+        ok = await message.reply(self._urls_to_string(vx_twtter_urls))
+
+        # Remove embeds from user message if reply is successful
+        if ok:
+            await message.edit(suppress=True)
+
+    async def _on_edit_twit_replacer(self, message_before: Message, message_after: Message):
+        self.logger.info("edit")
+        # skips if the message is sent by any bot
+        if not self._valid(message_after):
+            return
+
+        if not await self.config.guild(message_after.guild).get_attr(KEY_ENABLED)():
+            self.logger.info(
+                "VxTwit disabled for guild %s (%s), skipping",
+                message_after.guild.name,
+                message_after.guild.id,
+            )
+            return
+
+        video_embed_before = [embed for embed in message_before.embeds if embed.video]
+        video_embed_after = [embed for embed in message_after.embeds if embed.video]
+        new_video_embeds = [
+            embed for embed in video_embed_after if embed not in video_embed_before
         ]
```

Comment:
Thoughts on making these sets instead and then taking a set difference?
i.e. `new_video_embeds = set(video_embed_after) - set(video_embed_before)`?

Reply:
![image](https://github.com/SFUAnime/Ren/assets/75904609/d16b1baa-2537-49b1-b741-640c869da9c5)

Can't do that unfortunately, at least not with embed objects

Reply:
Ah okay, can disregard then.